pkg/client/client_test.go

```diff
@@ -0,0 +1,113 @@
+package client_test
+
+import (
+	"bufio"
+	"context"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/containers/conmon-rs/pkg/client"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+const maxRSSKB = 3200
+
+// TestConmonClient runs the created specs
+func TestConmonClient(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "ConmonClient")
+}
+
+var _ = Describe("ConmonClient", func() {
+	var pidFilePath, socketPath string
+	var sut *client.ConmonClient
+	BeforeEach(func() {
+		pidFile, err := ioutil.TempFile("", "pidfile")
+		Expect(err).To(BeNil())
+		pidFilePath = pidFile.Name()
+		socket, err := ioutil.TempFile("", "socket")
+		Expect(err).To(BeNil())
+		socketPath = socket.Name()
+	})
+
+	AfterEach(func() {
+		Expect(os.Remove(socketPath)).To(BeNil())
```

Comment:
should remove the pidFile as well

Reply:
pidfile is removed by the client itself. We could verify it's removed, or drop that behavior? I am unsure of it tbh